lib/diagnostics_channel.js

```diff
@@ -38,8 +38,7 @@ class ActiveChannel {
 
     // When there are no more active subscribers, restore to fast prototype.
     if (!this._subscribers.length) {
-      // eslint-disable-next-line no-use-before-define
-      ObjectSetPrototypeOf(this, Channel.prototype);
+      ObjectSetPrototypeOf(this, classDefinitions.Channel.prototype);
```

Comment:
this is still using something before it's defined - namely `classDefinitions`. If the linter isn't warning on this, it's a bug in the lint rule :-)

Reply:
I think it's not a bug in the rule but instead taking advantage of the way we have our rule configured:

```js
    'no-use-before-define': ['error', {
      classes: true,
      functions: false,
      variables: false,
    }],
```

We're only checking classes and not variables. However, I suspect the reason we're doing that isn't because it's totally cool to do it with classes but not variables. It's because there would be so much churn to make the rule comply if we turned it on for variables.

So, yeah, we shouldn't do this change. It doesn't make the code better--probably makes it slightly worse in terms of maintainability--and makes turning on checking for variables harder.

Reply:
So, should I close this PR @Trott ?

Reply:
> So, should I close this PR @Trott ?

Yes, I think so.

Reply:
Thanks.